src/open_inwoner/openzaak/clients.py

```diff
@@ -692,43 +702,207 @@ def fetch_open_tasks(self, bsn: str) -> list[OpenTask]:
+
+    client: TClient
+    result: Any
+    exception: Exception | None = None
+
+
+@dataclass(frozen=True)
+class MultiZgwClientProxyResult:
+    """Container for a multi-backend responses"""
+
+    responses: list[ZgwClientResponse]
+
+    @cached_property
+    def has_errors(self) -> bool:
+        return any(r.exception is not None for r in self.responses)
+
+    @cached_property
+    def failing_responses(self) -> list[ZgwClientResponse]:
+        return list(r for r in self if r.exception is not None)
+
+    @cached_property
+    def successful_responses(self) -> list[ZgwClientResponse]:
+        return list(r for r in self if r.exception is None)
+
+    def raise_on_failures(self):
+        """Raise a MultiZgwClientProxyError wrapping all errors raised by the clients."""
+        if not self.has_errors:
+            return
+
+        raise MultiZgwClientProxyError([r.exception for r in self.failing_responses])
+
+    def join_results(self):
+        """Joins results for all successful responses in a list."""
+        return list(
+            result for row in self.successful_responses for result in row.result
+        )
+
+    def __iter__(self):
+        yield from self.responses
+
+
+class MultiZgwClientProxy:
+    """A proxy to call the same method on multiple ZGW clients in parallel."""
+
+    clients: list[TClient] = []
+
+    def __init__(self, clients: list[TClient]):
+        self.clients = clients
+
+        if len(clients) == 0:
+            raise ValueError("You must specify at least one client")
+
+    def _call_method(self, method, *args, **kwargs) -> MultiZgwClientProxyResult:
+        if not all(hasattr(client, method) for client in self.clients):
+            raise AttributeError(f"Method `{method}` does not exist on the clients")
+
+        with parallel() as executor:
+            futures_mapping: Mapping[concurrent.futures.Future, TClient] = {}
+            for client in self.clients:
+                future = executor.submit(
+                    getattr(client, method),
+                    *args,
+                    **kwargs,
+                )
+                # Remember which future corresponds to which client,
+                # so we can associate them in the response
+                futures_mapping[future] = client
+
+            responses: list[ZgwClientResponse] = []
+            for task in concurrent.futures.as_completed(futures_mapping.keys()):
+                result: Any | None = None
+                exception: Exception = None
+                try:
+                    result: Any = task.result()
+                except Exception as exc:
+                    exception: Exception | None = task.exception()
+
+                responses.append(
+                    ZgwClientResponse(
+                        result=result, exception=exception, client=futures_mapping[task]
+                    )
+                )
+
+        return MultiZgwClientProxyResult(
+            responses=sorted(
```

Comment:
Why create a new list if the old one is not used? `responses.sort(..)` would work the same (and faster).